### PR TITLE
make entries with legacy class names backwards compatible

### PR DESCRIPTION
**Pull request by bdoyle0182**
fixes the issue where features of akka / pekko that are dependent on the persistence module and will potentially store the class name of an akka / pekko class as a part of the persisted data.

### CODE REVIEW

File: scala/src/main/scala/pekko/contrib/persistence/mongodb/MongoDataModel.scala [hunk 1]
```diff
@@ -48,17 +48,27 @@ case class Serialized[C <: AnyRef](bytes: Array[Byte],
 
   val hint = "ser"
   lazy val content: C = {
-
     val clazz = loadClass.getClassFor[X forSome { type X <: AnyRef }](className)
+    val backwardsCompatClazz = loadClass.getClassFor[X forSome { type X <: AnyRef }](className.replace("akka", "org.apache.pekko"))
```
Comment (pjfanning) [inline scala/src/main/scala/pekko/contrib/persistence/mongodb/MongoDataModel.scala:52]: Can you make this a def? Then it only runs if needed.

File: scala/src/main/scala/pekko/contrib/persistence/mongodb/MongoDataModel.scala [hunk 2]
```diff
@@ -48,17 +48,27 @@ case class Serialized[C <: AnyRef](bytes: Array[Byte],
 
   val hint = "ser"
   lazy val content: C = {
-
     val clazz = loadClass.getClassFor[X forSome { type X <: AnyRef }](className)
+    def backwardsCompatClazz = loadClass.getClassFor[X forSome { type X <: AnyRef }](className.replace("akka", "org.apache.pekko"))
```
Comment (thjaeckle) [inline scala/src/main/scala/pekko/contrib/persistence/mongodb/MongoDataModel.scala:52]: `className.replace("akka", "org.apache.pekko")` would replace any occurrences of `akka` ... 
Couldn't we check for `startsWith("akka.")` and if the class starts with "akka." - only then replace the first occurrence using `replaceFirst(regex, "org.apache.pekko")` with `org.apache.pekko`?

**Review by thjaeckle — CHANGES_REQUESTED** [review]
I would adjust the replacement of "akka" in classnames to make it more bulletproof

**Comment by bdoyle0182** [comment]
I cleaned it up and followed your suggestion good catch there to replaceFirst. I also realized we need to backport this to the old plugin to support roll backs, see this comment on the issue https://github.com/scullxbones/pekko-persistence-mongo/issues/10

**Review by thjaeckle — APPROVED** [review]

**Comment by thjaeckle** [comment]
Thanks for the contribution @bdoyle0182 👍 
Will release a `1.0.2` containing this fix.

**Comment by SakulK** [comment]
PSA for anyone that might hit the same issue as me: this fix unfortunately doesn't seem to address a similar issue with `PersistentFSM` snapshots. Here is the interesting part of the stacktrace that shows up in logs:
```
java.util.NoSuchElementException: None.get
     at scala.None$.get(Option.scala:627)
     at scala.None$.get(Option.scala:626)
     at pekko.contrib.persistence.mongodb.driver.ScalaDriverPersistenceSnapshotter$.deserializeSnapshot(ScalaDriverPersistenceSnapshotter.scala:43)
     at pekko.contrib.persistence.mongodb.driver.ScalaDriverPersistenceSnapshotter.$anonfun$findYoungestSnapshotByMaxSequence$5(ScalaDriverPersistenceSnapshotter.scala:82)
```
After digging deeper I found out that `deserialize` call here:
https://github.com/scullxbones/pekko-persistence-mongo/blob/b8127da3598eab0581d4f77cfc57b5c4b4ba6809/scala/src/main/scala/pekko/contrib/persistence/mongodb/driver/ScalaDriverPersistenceSnapshotter.scala#L44
returns:
```
Failure(java.io.NotSerializableException: Cannot find manifest class [akka.persistence.fsm.PersistentFSM$PersistentFSMSnapshot] for serializer with id [7].)
```
Which is thrown from https://github.com/apache/incubator-pekko/blob/main/actor/src/main/scala/org/apache/pekko/serialization/Serialization.scala#L246
I will share a workaround if I find one :wink:

**Comment by SakulK** [comment]
Ok, a hacky workaround that I came up with:
1. Make a copy of `org.apache.pekko.persistence.serialization.SnapshotSerializer` class from pekko persistence (here called `CustomSnapshotSerializer`)
2. In `snapshotFromBinary` replace the last line with:
```scala
    serialization
      .deserialize(snapshotBytes, serializerId, manifest)
      .recover {
        case _: NotSerializableException if manifest.startsWith("akka") =>
          serialization.deserialize(snapshotBytes, serializerId, manifest.replaceFirst("akka", "org.apache.pekko"))
      }
      .get
```
3. Register the serializer in the config, something like this:
```
pekko.actor {
  serialization-identifiers {
    "org.apache.pekko.persistence.serialization.SnapshotSerializer" = -8
    "org.apache.pekko.persistence.serialization.CustomSnapshotSerializer" = 8
  }
  serializers.customSnapshotSerializer = "org.apache.pekko.persistence.serialization.CustomSnapshotSerializer"
  serialization-bindings {
    "org.apache.pekko.persistence.serialization.Snapshot" = customSnapshotSerializer
  }
}
```
I'm not sure if it makes sense to propose the change from point 2. to the main pekko repo :thinking:

**Pull request merged**